enhancements/network/cloud-egress-ip.md

```diff
@@ -0,0 +1,445 @@
+plugin - currently - only looks at the subnet mask on the primary interface
+defined on the host, when determining if an egress IP can be assigned to a given
+OpenShift node.
+
+The following shows the API on GCP for updating an instance and assigning it an
+additional IP address
+
+```
+gcloud compute instances network-interfaces update $INSTANCE_NAME \
+    --zone $ZONE \
+    [--network-interface $NIC; default="nic0"]
+    --aliases "$RANGE_NAME:$RANGE_CIDR,RANGE_NAME:RANGE_CIDR,..."
+```
+
+where
+
+- `$ZONE` the zone that contains the instance.
+- `$NIC` is the name of the network interface to which you're adding an alias IP
+  address range.
+- `$RANGE_NAME` the name of the subnet secondary range from which to draw the
+  alias IP range.
+- `$RANGE_CIDR` is the IP range to assign to the interface. The range can be a
+  specific range (192.168.100.0/24), a single IP address (192.168.100.1), or a
+  network mask in CIDR format (/24). If the IP range is specified by network
+  mask only, the IP allocator chooses an available range with the specified
+  network mast and allocates it to the network interface.
+
+Note: an egress IP setup on GCP has already been accomplished using
+OVN-Kubernetes in a proof of concept / experimental mode. A couple of things
+were noticed during that experiment:
+
+- the alias IP assigned does not have any impact on the host, meaning: once the
+  alias IP is assigned using the GCP API, the IP address does not show up on the
+  host (on any interface) when doing `ip addr`
+- no additional host assignment of this alias IP is required for SNAT-ing to
+  occure to the alias IP, i.e: `ip addr add $ALIAS_IP dev eth0`
+- OVN's dedicated SNAT does not work, iptables rule were needed, such as:
+  `iptables -t nat -D POSTROUTING -p tcp -m tcp -d $NODE_IP/32 -j
+  SNAT--to-source $ALIAS_IP` to make things work. It is believed to be a bug in
+  OVN, but further coordination with them will be required to figure out why it
+  can't SNAT, whereas that iptables rule works. This is not a concern for
+  openshift-sdn, as it programs that exact iptables rule by default for egress
+  IP.
+
+[8]:
+https://cloud.google.com/sdk/gcloud/reference/compute/instances/network-interfaces/update
+
+#### Model
+
+##### network plugin <-> cloud API controller
+
+The network plugin needs to remain "master" of the egress IP setup. The reason
+for this is that the network plugin is the one and only component which health
+checks routes to egress nodes and re-balances the egress IP assignment depending
+on the result. This component will need to execute what the network plugin has
+determined is best.
+
+Given the above, it is however important that the network plugin is conveyed the
+cloud's information concerning the nodes subnet range. This is especially
+evident on GCP, as without it: the network plugin won't be able to perform any
+assignments at all, given the nodes default `/32` subnet.
```

Comment:
"node's" (twice here and some more later)

Reply:
In this case I was referencing "nodes" as in plural, but I guess that's: nodes' in correct English